src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -1301,23 +1294,27 @@ module Types = struct
              ; field "isTokenOwner" ~typ:bool
                  ~doc:"True if this account owns its associated token"
                  ~args:Arg.[]
-                 ~resolve:(fun _ { account; _ } ->
-                   match%map.Option account.token_permissions with
-                   | Token_owned _ ->
-                       true
-                   | Not_owned _ ->
-                       false )
+                 ~resolve:(fun { ctx = coda; _ } { account; _ } ->
+                   let open Option.Let_syntax in
+                   let%bind tip =
+                     coda |> Mina_lib.best_tip |> Participating_state.active
+                   in
+                   let ledger =
+                     Transition_frontier.Breadcrumb.staged_ledger tip
+                     |> Staged_ledger.ledger
+                   in
+                   let aid = account_id account in
+                   let%map token_owner =
+                     Ledger.token_owner ledger (Account_id.token_id aid)
+                   in
+                   Account_id.equal aid token_owner )
```

Comment:
This check will never be true, but always returning `false` may be misleading. Replace the whole body of this response with `None`.